File: src/com/mrpowergamerbr/loritta/utils/MathUtils.java
package com.mrpowergamerbr.loritta.utils;

public class MathUtils {
    public static double getPercentage(double val1, double val2) {
        return (val1 / val2) * 100;
    }
    
    public static boolean chance(final double e) {
        final double d = Math.random();
        return d < e / 100.0;
    }
}

File: loritta-discord/src/main/java/net/perfectdreams/loritta/utils/math/MathUtils.kt
package net.perfectdreams.loritta.utils.math

object MathUtils {
    /**
     * Evaluates [input] as a Math expression and returns the result
     *
     * Example: "2 + 2" will return 4
     *
     * @param  input             the input
     * @throws RuntimeException if the input is malformed
     * @return the result
     */
    fun evaluate(input: String) = MathParser(input).parse()

    /**
     * Evaluates [input] as a Math expression and returns the result, if the input is malformed, null will be returned.
     *
     * Example: "2 + 2" will return 4
     *
     * @see evaluate
     * @param  input             the input
     * @return the result
     */
    fun evaluateOrNull(input: String) = try { MathParser(input).parse() } catch (e: RuntimeException) { null }
}